pulsar-sql/presto-pulsar/src/main/java/org/apache/pulsar/sql/presto/PulsarMetadata.java

```diff
@@ -314,17 +365,43 @@ private ConnectorTableMetadata getTableMetadata(SchemaTableName schemaTableName,
         builder.addAll(getColumns(null, schema, new HashSet<>(), new Stack<>(), new Stack<>()));
 
         if (withInternalColumns) {
-            PulsarInternalColumn.getInternalFields().stream().forEach(new Consumer<PulsarInternalColumn>() {
-                @Override
-                public void accept(PulsarInternalColumn pulsarInternalColumn) {
-                    builder.add(pulsarInternalColumn.getColumnMetadata(false));
-                }
-            });
+            PulsarInternalColumn.getInternalFields().stream().forEach(pulsarInternalColumn ->
+                    builder.add(pulsarInternalColumn.getColumnMetadata(false)));
         }
-
-        return new ConnectorTableMetadata(schemaTableName, builder.build());
+        return builder.build();
     }
 
+    @VisibleForTesting
+    static Type convertPulsarType(SchemaType pulsarType) {
+        switch (pulsarType) {
+            case BOOLEAN:
+                return BooleanType.BOOLEAN;
+            case INT8:
+                return TinyintType.TINYINT;
+            case INT16:
+                return SmallintType.SMALLINT;
+            case INT32:
+                return IntegerType.INTEGER;
+            case INT64:
+                return BigintType.BIGINT;
+            case FLOAT:
+                return RealType.REAL;
+            case DOUBLE:
+                return DoubleType.DOUBLE;
+            case NONE:
+            case BYTES:
+                return VarbinaryType.VARBINARY;
+            case STRING:
+                return VarcharType.VARCHAR;
+            case DATE:
+            case TIME:
+            case TIMESTAMP:
+                return BigintType.BIGINT;
```

Comment:
Presto has date and time data types : https://prestodb.github.io/docs/current/language/types.html#date-and-time, i think it's better convert to date/time.